### PR TITLE
Fix email field validation error messages

### PR DESCRIPTION
**Pull request by mec**
If the user submits the email field with no value both the presence and
the format validation is triggered.

Switching the format validation to allow_blank stops this behaviour and
shows the appropriate message.

https://trello.com/c/YBl0fxWQ

### CODE REVIEW

File: spec/models/tslr_claim_spec.rb [hunk 1]
```diff
@@ -16,6 +16,7 @@
     it "validates that the value is in the correct format" do
       expect(build(:tslr_claim, email_address: "notan email@address.com")).not_to be_valid
       expect(build(:tslr_claim, email_address: "name@example.com")).to be_valid
+      expect(build(:tslr_claim, email_address: "")).to be_valid
```
Comment (pezholio) [inline spec/models/tslr_claim_spec.rb]: Actually, do we need another test to check `(build(:tslr_claim, email_address: nil)` is valid? (i.e. the default) I'm sure it is, but we probably need to check this assumption as if it isn't, the whole flow will break.

Comment (mec) [reply spec/models/tslr_claim_spec.rb]: I felt like we had enough coverage for that case in the validation contexts?

https://github.com/DFE-Digital/dfe-teachers-payment-service/blob/801f43202962a1a42dbc9a176f74898ad93c4866/spec/models/tslr_claim_spec.rb#L252

Comment (pezholio) [reply spec/models/tslr_claim_spec.rb]: Ah, true dat. Carry on then :)
